src/omnicore/rpc.cpp

```diff
@@ -968,87 +964,119 @@ Value gettradessince_MP(const Array& params, bool fHelp)
+/* TODO: point for discussion - "trade history" is ambiguous; providing trade history for a
+ * pair is likely to be more useful than trade history for an address.  Regardless the existing
+ * function did neither (only showing open trades, not trade history) so has been rewritten.
+ *
+ * Perhaps consider renaming this to gettradehistoryforaddress_OMNI to match existing naming
+ * (eg getallbalancesforaddress_MP) then look at addition of new gettradehistoryforpair_OMNI
+ * call.
+ */
+Value gettradehistoryforaddress_OMNI(const Array& params, bool fHelp)
 {
-   if (fHelp || params.size() < 1)
+    if (fHelp || params.size() < 1 || params.size() > 3)
         throw runtime_error(
             "gettradehistory_MP\n"
-            "\nAllows user to retreive trade history for some address on the Metadex\n"
-            
+            "\nAllows user to retrieve MetaDEx trade history for the supplied address\n"
             "\nArguments:\n"
-            "1. address          (string, required) address to query history on\n"
-            "2. number            (int, optional ) number of trades to retreive\n"
-            "3. property_id         (int, optional) filter by propertyid on one side\n"
+            "1. address          (string, required) address to retrieve history for\n"
+            "2. count            (int, optional) number of trades to retrieve (default: 10)\n"
+            "3. propertyid       (int, optional) filter by propertyid transacted\n"
         );
 
-  string address = params[0].get_str();
-  unsigned int number_trades = (params.size() == 2 ? (unsigned int) params[1].get_int64() : 512);
-  unsigned int propertyIdFilter = 0;
+    Array response;
+
+    string address = params[0].get_str();
+    int64_t count = 10;
+    if (params.size() > 1) {
+        count = params[1].get_int64();
+        if (count < 1) // TODO: do we need a maximum here?
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid value for count parameter");
+    }
+    uint32_t propertyId = 0;
+    if (params.size() > 2) {
+        int64_t tmpPropertyId = params[2].get_int64();
+        if (tmpPropertyId < 1 || tmpPropertyId > 4294967295) // not safe to do conversion
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid property identifier");
+        propertyId = boost::lexical_cast<uint32_t>(tmpPropertyId);
+        CMPSPInfo::Entry sp;
+        if (false == _my_sps->getSP(propertyId, sp))
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Property identifier does not exist");
+    }
 
-  bool filter_by_one = (params.size() == 3) ? true : false;
+    // to provide a proper trade history for an address we must utilize a two part methodology:
+    // 1) check the metadex maps for open trades
+    // 2) check the trades database for closed trades
+
+    // vector vecTransactions will hold the txids from both stage 1 and stage 2 to avoid duplicates
+    std::vector<uint256> vecTransactions;
+
```

Comment:
You may "open a section" here with `{` and `LOCK(cs_tally);`, which should be closed after `t_tradelistdb->...` and before the other loop with `populateRPC..`.

Removing the `threadSafe` flag in `rpcserver.cpp` can be done, unrelated to the tally lock, because neither the wallet, nor globals of `main.cpp` are used here, as far as I can see.